scripts/packager-codes/si-packagers-refresh.py

```diff
@@ -0,0 +1,214 @@
+	input_address = "<>".join(input_address.split('\r'))
+
+	# fetch last occurence
+	# words 123A, place, 4567 city name
+	# Á found in a city name (PROSENJAKOVCI -PÁRTOSFALVA)
+	pattern = r'(([a-zčćžđšA-ZČĆŽĐŠŽ\s\-\.]+\d+[ABCDEFGIJ]?),(?:[a-zčćžđšA-ZČĆŽĐŠŽ\s\-\.\<\>]+,\s*)?[\<\>]*(\s*\d{4}[a-zčćžđšA-ZČĆŽĐŠŽÁ\s\-\.\<\>]+)$)'
+
+	match = re.search(pattern, input_address)
+
+	
+	if match:
+		output_address = (f"{match.group(2).strip().title()}, {match.group(3).replace('<>', ' ').strip().title()}")
+	else:
+		# MOŠNJE , MOŠNJE, 4240 RADOVLJICA -> no street number (also DIJAŠKA ULICA , 5220 TOLMIN)
+		# instead, fetch "something, postal_code city"
+		pattern_2 = r'(([a-zčćžđšA-ZČĆŽĐŠŽ\s\-\.]+),(\s*\d{4})([a-zčćžđšA-ZČĆŽĐŠŽÁ\s\-\.\<\>]+)$)'
+		match_2 = re.search(pattern_2, input_address)
+		if match_2:
+			output_address = (f"{match_2.group(2).strip().title()}, {match_2.group(3).replace('<>', ' ').strip().title()}")		
+		else:
+			print("Match problem", input_address)
+			output_address = input_address
+
+	return output_address
+
+
+def convert_address_to_lat_lng(address_to_convert: str) -> str:
+	# free plan: 1 request per second
+	sleep(1)
+
+	print("address_to_convert: ", address_to_convert)
+	street, post_and_town = address_to_convert.split(',')
+	postalcode = post_and_town.strip().split()[0]
+	town = " ".join(post_and_town.strip().split()[1:])
+
+	url = f"https://geocode.maps.co/search?street={street}&town={town}&postalcode={postalcode}&country=Slovenia&country_code=si&api_key={api_key}"
+
+	try:
+		response = requests.get(url)
+		data = response.json()
+		if data != []:
+			lat_lng = f"{data[0]['lat']},{data[0]['lon']}"
+		else:
+			sleep(1)
+			# drop housenumber (example: Vrhpolje 1D, 5271 Vipava)
+			url_2 = f"https://geocode.maps.co/search?street={' '.join(street.split()[:-1])}&town={town}&postalcode={postalcode}&country=Slovenia&country_code=si&api_key={api_key}"
+			
+			try:
+				response = requests.get(url_2)
+				data = response.json()
+				if data != []:
+					lat_lng = f"{data[0]['lat']},{data[0]['lon']}"
+				else:
+					sleep(1)
+					# drop street (example: Gabrovlje 14, 3214 Zreče)
+					url_3 = f"https://geocode.maps.co/search?town={town}&postalcode={postalcode}&country=Slovenia&country_code=si&api_key={api_key}"
+					
+					try:
+						response = requests.get(url_3)
+						data = response.json()
+						if data != []:
+							lat_lng = f"{data[0]['lat']},{data[0]['lon']}"
+						else:
+							print(f'Empty response for: {address_to_convert}')
+							sys.exit(1)
+					except (requests.exceptions.RequestException, KeyError, IndexError) as e:
+						print(f"Error: {e}")
+						sys.exit(1)
+			except (requests.exceptions.RequestException, KeyError, IndexError) as e:
+				print(f"Error: {e}")
+				sys.exit(1)
+	except (requests.exceptions.RequestException, KeyError, IndexError) as e:
+		print(f"Error: {e}")
+		sys.exit(1)
+
+	return lat_lng
+
+
+
+file_name = "slovenian_packaging_raw.csv"
```

Comment:
It's always good to put such sections in a `if __name__ == "__main__"`

Reply:
I applied suggestions